File: Data Structures & Algorithms/intersection-of-two-arrays/submission-0.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> uset; 
        vector<int> arr;
        for(int i = 0; i < nums1.size();i++){
            uset.insert(nums1[i]);
        }
        for(int i = 0; i < nums2.size();i++){
            if(uset.count(nums2[i])) arr.push_back(nums2[i]);
        }
        return arr;
    }
};

File: Data Structures & Algorithms/merge-sorted-array/submission-0.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        size = m + n - 1
        m = m - 1
        n = n - 1
        while size >= 0 and m >= 0 and n >= 0:
            if nums1[m] > nums2[n]:
                nums1[size] = nums1[m]
                m -= 1
            elif nums2[n] >= nums1[m]:
                nums1[size] = nums2[n]
                n -= 1
            size -= 1
        while n >= 0 and size >= 0:
            nums1[size] = nums2[n]
            n -=1
            size -=1
